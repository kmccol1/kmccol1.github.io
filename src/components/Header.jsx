//Filename: Header.jsx
//Author: Kyle McColgan
//Date: 9 May 2025
//Description: Contains the React component for the website contact header section.

import React from 'react';
import profilePic from '/images/kdm-ocean-01.png';

function Header() {
    return (
        <header>
            <h1>Kyle D. McColgan</h1>
            <h2>Technology Professional</h2>
            <img className="profile-img"
            src={profilePic}
            alt="KDM picture" />
        </header>
    );
}

export default Header;
