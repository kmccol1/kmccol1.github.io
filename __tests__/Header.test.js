// __tests__/Header.test.js
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../src/components/Header';

it('renders the header with the correct title', () => {
    render(<Header />);
    const title = screen.getByRole('heading', { level: 1 });
    expect(title).toHaveTextContent('Kyle D. McColgan');
});

it('renders the header with the correct subtitle', () => {
    render(<Header />);
    const subtitle = screen.getByRole('heading', { level: 2 });
    expect(subtitle).toHaveTextContent('Technology Professional');
});

it('renders the header image', () => {
    render(<Header />);
    const image = screen.getByAltText('KDM picture');
    expect(image).toBeInTheDocument();
});

it('has the correct image source', () => {
    render(<Header />);
    const image = screen.getByAltText('KDM picture');
    expect(image).toHaveAttribute('src', 'test-file-stub');
});

it('has the correct alt text for the image', () => {
    render(<Header />);
    const image = screen.getByAltText('KDM picture');
    expect(image).toHaveAttribute('alt', 'KDM picture');
});

it('renders the image with the correct class', () => {
    render(<Header />);
    const image = screen.getByAltText('KDM picture');
    expect(image).toHaveClass('profile-img');
});

it('renders the header element', () => {
    render(<Header />);
    const header = screen.getByRole('banner');
    expect(header).toBeInTheDocument();
});

it('does not log any errors to the console', () => {
    const consoleError = jest.spyOn(console, 'error').mockImplementation();
    render(<Header />);
    expect(consoleError).not.toHaveBeenCalled();
});

it('ensures the header is not empty', () => {
    render(<Header />);
    const header = screen.getByRole('banner');
    expect(header).not.toBeEmptyDOMElement();
});

it('matches the snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
});
