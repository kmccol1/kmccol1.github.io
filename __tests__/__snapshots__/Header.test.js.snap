// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <header>
    <h1>
      Kyle D. McColgan
    </h1>
    <h2>
      Technology Professional
    </h2>
    <img
      alt="KDM picture"
      class="profile-img"
      src="test-file-stub"
    />
  </header>
</DocumentFragment>
`;
